homeassistant/components/homekit/type_switches.py

```diff
@@ -71,21 +73,48 @@ def __init__(self, *args):
         self._domain = split_entity_id(self.entity_id)[0]
         self._flag_state = False
 
+        self.activate_only = self.is_activate(
+            self.hass.states.get(self.entity_id))
+
         serv_switch = self.add_preload_service(SERV_SWITCH)
         self.char_on = serv_switch.configure_char(
             CHAR_ON, value=False, setter_callback=self.set_state)
 
+    def is_activate(self, state):
+        """Check if entity is activate only."""
+        can_cancel = state.attributes.get(ATTR_CAN_CANCEL)
+        if self._domain == 'script' and not can_cancel:
+            return True
+        return False
```

Comment:
`scene` support could be added here

Reply:
As you can see in my comment above, I already added it. Works fine too